### PR TITLE
feat: [STO-4441]: STO core client implementation

### PR DESCRIPTION
**Pull request by r0610205**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)

### CODE REVIEW

**Review by williamwissemann — APPROVED** [review]

**Review by williamwissemann — APPROVED** [review]
> Reviewed 30 of 30 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: :shipit: complete! ✓

File: 320-ci-execution/src/test/java/io/harness/stateutils/buildstate/K8BuildSetupUtilsTest.java [hunk 1]
```diff
@@ -142,9 +148,16 @@ public void shouldCreatePodParameters() throws Exception {
 
     String tiEndpoint = "http://localhost:8078";
     String tiToken = "token";
+
+    String stoEndpoint = "http://localhost:8078";
+    String stoToken = "stoToken";
```
Comment (chris-harness) [inline 320-ci-execution/src/test/java/io/harness/stateutils/buildstate/K8BuildSetupUtilsTest.java:153]: Should this port be different?

Comment (r0610205) [reply 320-ci-execution/src/test/java/io/harness/stateutils/buildstate/K8BuildSetupUtilsTest.java:153]: That's a test, I can update it for consistency

**Review by williamwissemann — APPROVED** [review]
> Reviewed 5 of 8 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: 1. unresolved discussions.

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 19 of 30 files at r1, 5 of 8 files at r3, 7 of 7 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: 1. unresolved discussions.

**Review by jennyjames21 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: 1. unresolved discussions.

**Review by jennyjames21 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: 1. unresolved discussions.

**Comment by r0610205** [comment]
*[`320-ci-execution/src/test/java/io/harness/stateutils/buildstate/K8BuildSetupUtilsTest.java` line 153 at r2](https://reviewable.io/reviews/harness/harness-core/32742#-N1Aihy_hbgsAiKNNABk-r2-153:-N1L0kPo0tiiiZ38JXJz:b-896fjx) ([raw file](https://github.com/harness/harness-core/blob/eeeea620bd8c3f1e7afb0a40a29d32de358235b2/320-ci-execution/src/test/java/io/harness/stateutils/buildstate/K8BuildSetupUtilsTest.java#L153)):*
<details><summary><i>Previously, r0610205 (Sergey Bobrov) wrote…</i></summary><blockquote>

That's a test, I can update it for consistency
</blockquote></details>

Done

**Review by atmollo-harness — APPROVED** [review]

**Review by chris-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: :shipit: complete! ✓

**Review by hextechpal — APPROVED** [review]
> Reviewed 18 of 30 files at r1, 5 of 8 files at r3, 4 of 7 files at r4, 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: :shipit: complete! ✓

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 4 of 4 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32742)* status: :shipit: complete! ✓

**Comment by r0610205** [comment]
trigger ti3

**Pull request merged**